Ginger/GingerCore/ALM/Qtest/ExportToQtest.cs

```diff
@@ -0,0 +1,594 @@
+        /// </summary>
+        /// <param name="activitiesGroup">Activities Group to Export</param>
+        /// <param name="mappedTest">The QC Test Case which mapped to the Activities Group (in case exist) and needs to be updated</param>
+        /// <param name="uploadPath">Upload path in QC Test Plan</param>
+        /// <param name="result">Export error result</param>
+        /// <returns></returns>
+        public static bool ExportActivitiesGroupToQC(ActivitiesGroup activitiesGroup, Test mappedTest, string uploadPath, ObservableList<ExternalItemFieldBase> testCaseFields, ref string result)
+        {
+            Test test;
+
+            try
+            {
+                if (mappedTest == null)
+                {
+                    //##create new Test Case in QC
+                    TestFactory TestF = (TestFactory)mTDConn.TestFactory;
+                    test = (Test)TestF.AddItem(System.DBNull.Value);
+                    test.Type = "MANUAL";
+
+                    //set the upload path
+                    TreeManager treeM = (TreeManager)mTDConn.TreeManager;
+                    ISysTreeNode testParentFolder = (ISysTreeNode)treeM.get_NodeByPath(uploadPath);
+                    test["TS_SUBJECT"] = testParentFolder.NodeID;
+                    
+                }
+                else
+                {
+                    //##update existing test case
+                    test = ImportFromQC.GetQCTest(activitiesGroup.ExternalID);
+
+                    //delete the un-needed steps
+                    DesignStepFactory stepF = test.DesignStepFactory;
+                    List stepsList = stepF.NewList("");
+                    foreach (DesignStep step in stepsList)
+                    {
+                        if (activitiesGroup.ActivitiesIdentifiers.Where(x => x.IdentifiedActivity.ExternalID == step.ID.ToString()).FirstOrDefault() == null)
+                            stepF.RemoveItem(step.ID);
+                    }
+
+                    //delete the existing parameters
+                    StepParams testParams = test.Params;
+                    if (testParams.Count > 0)
+                    {
+                        for (int indx = 0; indx < testParams.Count; indx++)
+                        {
+                            testParams.DeleteParam(testParams.ParamName[indx]);
+                            testParams.Save();
+                        }
+                    }
+
+                }
+
+                //set item fields
+                foreach (ExternalItemFieldBase field in testCaseFields)
+                {
+                    if (field.ToUpdate || field.Mandatory)
+                    {
+                        if (string.IsNullOrEmpty(field.SelectedValue) == false && field.SelectedValue != "NA")
+                            test[field.ID] = field.SelectedValue;
+                        else
+                            try { test[field.ID] = "NA"; }
+                            catch { }
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Handle the exception or explain in a comment why it can be ignored.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5395796)